PocketKit/Tests/PocketKitTests/Refresh/HomeRefreshCoordinatorTests.swift

```diff
@@ -0,0 +1,134 @@
+    private var userDefaults: UserDefaults!
+    private var source: MockSource!
+    private var subscriptions: Set<AnyCancellable>!
+
+    override func setUpWithError() throws {
+        notificationCenter = NotificationCenter()
+        userDefaults = UserDefaults()
+        source = MockSource()
+        subscriptions = []
+    }
+
+    override func tearDownWithError() throws {
+        userDefaults.removeObject(forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+    }
+
+    func subject(
+        notificationCenter: NotificationCenter? = nil,
+        userDefaults: UserDefaults? = nil,
+        source: Source? = nil,
+        refreshRuleTime: Int = 1
+    ) -> HomeRefreshCoordinator {
+        HomeRefreshCoordinator(
+            notificationCenter: notificationCenter ?? self.notificationCenter,
+            userDefaults: userDefaults ?? self.userDefaults,
+            source: source ?? self.source,
+            refreshRuleTime: refreshRuleTime
+        )
+    }
+
+    func test_firstRefresh_setsUserDefaults() {
+        XCTAssertNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+        source.stubFetchSlateLineup { _ in }
+
+        let expectRefresh = expectation(description: "Refresh home")
+
+        let coordinator = subject()
+
+        coordinator.refresh {
+            expectRefresh.fulfill()
+        }
+        wait(for: [expectRefresh], timeout: 1)
+        XCTAssertNotNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+    }
+
+    func test_coordinator_whenDataIsNotStale_doesNotRefreshHome() {
+        let date = Date()
+        userDefaults.setValue(date, forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+        XCTAssertNotNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+        source.stubFetchSlateLineup { _ in }
+
+        let coordinator = subject()
+        coordinator.refresh {
+            XCTFail("Should not have refreshed")
+        }
+        XCTAssertEqual(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey) as? Date, date)
+    }
+
+    func test_coordinator_whenDataIsStale_refreshesHome() {
+        let date = Calendar.current.date(byAdding: .hour, value: -2, to: Date())
+        userDefaults.setValue(date, forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+        XCTAssertNotNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+        source.stubFetchSlateLineup { _ in }
+
+        let coordinator = subject()
+
+        let expectRefresh = expectation(description: "Refresh home")
+        coordinator.refresh {
+            expectRefresh.fulfill()
+        }
+        wait(for: [expectRefresh], timeout: 1)
+        XCTAssertNotEqual(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey) as? Date, date)
+    }
+
+    func test_coordinator_whenForceRefresh_refreshesHome() {
+        XCTAssertNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
```

Comment:
Do we want this to be nil? I feel like the code path here is if that there _is_ no last date, then we're not forcing it, we're just hitting the case that we've never been refreshed. In my mind, if we set this value to `Date()`, _then_ attempt a force-refresh, we should refresh (since we're within the 12 hours, still).

Reply:
yes, you are right! good catch! I set the date up to be the current date, so that a forced refresh will call the refresh button even if within the 12 hours.